yourbench/utils/prompts.py

```diff
@@ -0,0 +1,497 @@
+
+```python
+class QuestionAnswerPair(BaseModel):
+    thought_process: str # Explanation of integrative reasoning and rationale
+    question_type: Literal["analytical", "application-based", "clarification",
+                           "counterfactual", "conceptual", "true-false",
+                           "factual", "open-ended", "false-premise", "edge-case"]
+    question: str
+    answer: str
+    estimated_difficulty: int  # 1-10, moderately challenging as per additional instructions
+    citations: List[str]  # Exact supporting quotes from text_chunks
+```
+
+## Output Format
+
+First, thoroughly conduct your analysis within `<document_analysis>` XML tags. Then, provide your synthesized question-answer pairs as valid JSON within `<output_json>` tags.
+
+## Important Notes
+- Prioritize depth and thoughtfulness in your reasoning paths.
+- Allow natural complexity to guide question formulation, aiming for moderate challenge.
+- Precisely cite verbatim excerpts from text chunks.
+- Clearly communicate your thought process for integrative reasoning.
+- Adhere strictly to JSON formatting and Pydantic validation requirements.
+- Generate questions that genuinely inspire deeper reflection or meaningful exploration of the provided content.
+- When generating questions, NEVER include phrases like 'as per the text,' 'according to the document,' or any similar explicit references. Questions should inherently integrate content naturally and stand independently without explicit references to the source material
+
+"""
+
+MULTI_HOP_QUESTION_GENERATION_USER_PROMPT = """<title>
+{title}
+</title>
+
+<document_summary>
+{document_summary}
+</document_summary>
+
+<text_chunks>
+{chunks}
+</text_chunks>
+
+<additional_instructions>
+{additional_instructions}
+</additional_instructions>"""
+
+
+ZEROSHOT_QA_USER_PROMPT = """Answer the following question:
+
+<question>
+{question}
+</question>
+
+Enclose your full answer in <answer> XML tags. For example:
+
+<answer>
+[your answer here]
+</answer>"""
+
+GOLD_QA_USER_PROMPT = """Answer the following question:
+
+<question>
+{question}
+</question>
+
+Here is a summary of the document the question is asked from which may be helpful:
+
+<document_summary>
+{summary}
+</document_summary>
+
+And here is a relevant chunk of the document which may prove useful
+
+<document>
+{document}
+</document>
+
+Enclose your full answer in <answer> XML tags. For example:
+
+<answer>
+[your answer here]
+</answer>"""
+
+JUDGE_ANSWER_SYSTEM_PROMPT = """You will be provided with the summary of a document, a piece of text, a question generated from that text, and the correct or "gold" answer to the question. Additionally, you will receive two answers: Answer A and Answer B. Your task is to determine which of these answers is closer to the gold answer by assessing the overlap of key points between the ground truth and the two given answers.
```

Comment:
We might want to have a judge working on generations for one single model instead of arena style

Reply:
So a prompt adapted to this will be needed imo

Reply:
we can perhaps just add a new prompt for lighteval style judging

Reply:
Can be managed in lighteval for now